alerting/src/main/kotlin/org/opensearch/alerting/AlertingV2Utils.kt

```diff
@@ -0,0 +1,159 @@
+import org.opensearch.transport.client.node.NodeClient
+
+object AlertingV2Utils {
+    // Validates that the given scheduled job is a Monitor
+    // returns the exception to pass into actionListener.onFailure if not.
+    fun validateMonitorV1(scheduledJob: ScheduledJob): Exception? {
+        if (scheduledJob is MonitorV2) {
+            return IllegalStateException("The ID given corresponds to a V2 Monitor, but a V1 Monitor was expected")
+        } else if (scheduledJob !is Monitor && scheduledJob !is Workflow) {
+            return IllegalStateException("The ID given corresponds to a scheduled job of unknown type: ${scheduledJob.javaClass.name}")
+        }
+        return null
+    }
+
+    // Validates that the given scheduled job is a MonitorV2
+    // returns the exception to pass into actionListener.onFailure if not.
+    fun validateMonitorV2(scheduledJob: ScheduledJob): Exception? {
+        if (scheduledJob is Monitor || scheduledJob is Workflow) {
+            return IllegalStateException("The ID given corresponds to a V1 Monitor, but a V2 Monitor was expected")
+        } else if (scheduledJob !is MonitorV2) {
+            return IllegalStateException("The ID given corresponds to a scheduled job of unknown type: ${scheduledJob.javaClass.name}")
+        }
+        return null
+    }
+
+    // appends user-defined custom trigger condition to PPL query, only for custom condition Triggers
+    fun appendCustomCondition(query: String, customCondition: String): String {
+        return "$query | $customCondition"
+    }
+
+    fun appendDataRowsLimit(query: String, maxDataRows: Long): String {
+        return "$query | head $maxDataRows"
+    }
+
+    // returns PPL query response as parsable JSONObject
+    suspend fun executePplQuery(query: String, client: NodeClient): JSONObject {
+        // call PPL plugin to execute query
+        val transportPplQueryRequest = TransportPPLQueryRequest(
+            query,
+            JSONObject(mapOf("query" to query)),
+            null // null path falls back to a default path internal to SQL/PPL Plugin
+        )
+
+        val transportPplQueryResponse = PPLPluginInterface.suspendUntil {
+            this.executeQuery(
+                client,
+                transportPplQueryRequest,
+                it
+            )
+        }
+
+        val queryResponseJson = JSONObject(transportPplQueryResponse.result)
+
+        return queryResponseJson
+    }
+
+    // searches a given custom condition eval statement for the name of
+    // the eval result variable and returns it
+    fun findEvalResultVar(customCondition: String): String {
+        // the PPL keyword "eval", followed by a whitespace must be present, otherwise a syntax error from PPL plugin would've
+        // been thrown when executing the query (without the whitespace, the query would've had something like "evalresult",
+        // which is invalid PPL
+        val regex = """\beval\s+([a-zA-Z_][a-zA-Z0-9_]*)\s*=""".toRegex()
```

Comment:
can this be a static variable and given our dependency on sql plugin, can we import invalid ppl regex from there?

Reply:
can we also break this down into substrings that are stored in variables like EVAL, STARTS_WITH_REGEX, ALPHANUMERIC_REGEX, WILDCARD_REGEX etc. so it its more readable